src/service/csmService.js

```diff
@@ -69,6 +69,14 @@ class CsmService {
             csmConfig.widgetType : this.widgetType;
     }
 
+    _shouldExecuteCSM() {
+        if (typeof csm === 'undefined') {
```

Comment:
if csm service is disabled, no logic in this csm service should be executed. Why are you adding so many branch here instead of just 1 branch at the entrypoint of this csm Service

Reply:
Each method will access `csm.Metric` and trigger `catch` block. Alternate solution is to refactor more code, but trying to make minimal changes and not break existing logic.